cmake/sip_helper.cmake

```diff
@@ -79,14 +79,19 @@ function(build_sip_binding PROJECT_NAME SIP_FILE)
     set(LIBRARY_DIRS ${${PROJECT_NAME}_LIBRARY_DIRS})
     set(LDFLAGS_OTHER ${${PROJECT_NAME}_LDFLAGS_OTHER})
 
+    set(EXTRA_DEFINES "")
+    if(DEFINED BUILD_SHARED_LIBS)
```

Comment:
`DEFINED` doesn't seem sufficient to check. This condition would be `TRUE` even if the CMake variable `BUILD_SHARED_LIBS` is `FALSE`.

Reply:
Thank for the catch. I corrected the check now.

Reply:
Without the `DEFINED` check the code is accessing a potentially undefined variable (which results in a CMake warning if that warning is enabled). Please keep the old check with `DEFINED` as well as the new check for testing its actual value.

Reply:
Got it. I added both as the check.